packages/experimental/Button/src/Button.tsx

```diff
@@ -28,7 +28,8 @@ export const buttonLookup = (layer: string, state: IPressableState, userProps: B
     layer === userProps['shape'] ||
     (!userProps['shape'] && layer === 'rounded') ||
     (layer === 'hasContent' && !userProps.iconOnly) ||
-    (layer === 'hasIcon' && (userProps.icon || userProps.loading))
+    (layer === 'hasIconAfter' && (userProps.icon || userProps.loading) && userProps.iconPosition === 'after') ||
+    (layer === 'hasIconBefore' && (userProps.icon || userProps.loading) && (!userProps.iconPosition || userProps.iconPosition === 'before'))
```

Comment:
Is it possible to run into a state where you can have both `haveIconBefore` and `userProps.iconPosition === 'after'` ?

Reply:
It shouldn't be, since iconPosition can only be before, after, or undefined.

Reply:
I guess the question was more, could we run into a state where the user provided prop and the layer string are out of sync?

Reply:
I'm a little confused. The way the lookup function works is that it takes the list of states that we tell the component it can have and runs through each one asking if the state applies. Based on the states that returned true, the component will look at the token definitions to apply the correct styles for that state.

So this part of the condition basically asks if we're checking for the hasIconBefore state/condition, and if so, returns true is iconPosition is either undefined or set to "before". Hopefully that clears things up?

Reply:
Oh, yes I misunderstood this function derp. It's going through every layer, so we'll never hit that case. My bad!